### ISSUE TITLE
Can't complete initial setup in Canada

### ISSUE THREAD

**Issue by clarencetso** [Hyundai-Kia-Connect/kia_uvo#993]
**Region and Brand of car**
Canada
Ioniq 5

**Version of the integration**
2.31.0

**Describe the bug**
Initial setup fails with Unexpected error after inputting auth credentials

```
This error originated from a custom integration.

Logger: custom_components.kia_uvo.config_flow
Source: custom_components/kia_uvo/config_flow.py:160
integration: kia_uvo (documentation, issues)
First occurred: 11:28:52 AM (3 occurrences)
Last logged: 11:38:44 AM

Unexpected exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/requests/models.py", line 974, in json
    return complexjson.loads(self.text, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/simplejson/__init__.py", line 514, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/simplejson/decoder.py", line 386, in decode
    obj, end = self.raw_decode(s)
               ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/simplejson/decoder.py", line 416, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/kia_uvo/config_flow.py", line 160, in async_step_user
    await validate_input(self.hass, user_input)
  File "/config/custom_components/kia_uvo/config_flow.py", line 64, in validate_input
    token: Token = await hass.async_add_executor_job(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/hyundai_kia_connect_api/KiaUvoApiCA.py", line 137, in login
    response = response.json()
               ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/requests/models.py", line 978, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

Outlined here: https://github.com/fuatakgun/kia_uvo/blob/master/README.md

**To Reproduce**
Set up integration, input credentials on configuration screen, login.

**Expected behavior**
Successful login

**Additional context**
1. I enabled the debug log in `configuration.yaml` per instructions before producing the above logs
2. I tried inputting incorrect credentials on purpose, which produced a `AuthenticationError` instead. This is how I know that I am inputting the correct credentials.

**Comment by cdnninja** [comment]
Are all cars on the account checking in and working?

**Comment by Phoenix260** [comment]
I have the same issue.
Kona EV in Canada
Deleted and reinstalled due to integration issues, now I can't setup.

**Comment by cdnninja** [comment]
Please answer the above. As well providing debug logs as per issue template would help.

**Comment by clarencetso** [comment]
There's only one car on the account, and it is checking in and working in the bluelink app.

Re: debug logs, I didn't know you had to click into Configuration --> Logs --> **show raw logs** in order to see the debug output. Here it is. From this error, any ideas why the Bluelink Server would detect a US IP which is certainly not my Canada IP? I'm not on any VPNs.

```
2024-12-21 16:55:58.375 DEBUG (SyncWorker_17) [hyundai_kia_connect_api.KiaUvoApiCA] hyundai_kia_connect_api - Sign In Response <br>
<br>
<center>
<h2>
The request / response that are contrary to the Web firewall security policies have been blocked.
</h2>
<table>
<tr>
<td>Detect time</td>
<td>2024-12-21 13:55:56</td>
</tr>
<tr>
<td>Detect client IP</td>
<td>172.27.80.11</td>
</tr>
<tr>
<td>Detect URL</td>
<td>http://mybluelink.ca/tods/api/lgn</td>
</tr>
</table>
</center>
<br>
```

**Comment by cdnninja** [comment]
That's not a USA IP.  It's an internal IP strange it's exposed to Hyundai Kia.   Any unique network setup?

**Comment by clarencetso** [comment]
Ah, weird, some ip lookup site said it was in USA, but never mind, my bad. 

I didn't think my network was particularly unique. I installed HA OS using the VirtualBox image at https://www.home-assistant.io/installation/linux (bridged networking) on an Ubuntu host. I checked all the IPs in the docker containers in the HA machine, the Ubuntu host, and none of the 172 IPs were 172.27.*. Super strange.

**Comment by clarencetso** [comment]
Based on https://github.com/Hyundai-Kia-Connect/kia_uvo/issues/860 I tried changing my password, and that resolved the issue. As part of changing the password, I decided to remove symbols from it as well, since sometimes that can cause problems. However, it's unclear whether just a password change vs. removing the symbols is what fixed the issue.

Thanks for looking into this initially for me - closing the issue.
